scale-tests/batch_test.py

```diff
@@ -73,15 +78,43 @@ def _is_strict(dispatcher: typing.Dict) -> bool:
     return dispatcher["service"].get("service_account") is not None
 
 
-def submit_job(dispatcher: typing.Dict, duration: int, config: typing.List[str]):
+def _get_gpu_user_conf(args):
+    def _verify_required_args():
+        if not (args["--spark-mesos-max-gpus"] and
+                args["--spark-mesos-executor-gpus"] and
+                args["--docker-image"]):
+            log.error("""
+            Missing required arguments for running gpu jobs. Please include:
+            --spark-mesos-max-gpus
+            --spark-mesos-executor-gpus
+            --docker-image
+            """)
+
+    _verify_required_args()
+
+    # Based on testing, 20gb per GPU is needed to run the job successfully.
+    # This is due to memory being divvied up and allocated to each GPU device.
+    memory_multiplier = 20
+    memory = int(args["--spark-mesos-executor-gpus"]) * memory_multiplier
+    return ["--conf", "spark.driver.memory={}g".format(str(memory)),
+            "--conf", "spark.executor.memory={}g".format(str(memory)),
+            "--conf", "spark.cores.max={}".format(args["--spark-cores-max"]),
```

Comment:
`spark.cores.max` is already set as part of the "standard" config options.

Reply:
you mean in actual Spark's configuration? It doesn't look like it: https://spark.apache.org/docs/latest/configuration.html#scheduling

unless you meant its set somewhere else?